src/network/networktool.cpp

```diff
@@ -0,0 +1,178 @@
+#include "logger.h"
+#include "maddy/parser.h"
+#include "networkmanager.h"
+#include "networktool.h"
+#include "notificationmanager.h"
+
+#include <QJsonArray>
+#include <QJsonDocument>
+#include <QRegularExpression>
+#include <QTimer>
+#include <QUrl>
+
+PING_LOGGING_CATEGORY(NETWORKTOOL, "ping.networktool")
+
+void NetworkTool::checkInterfaceUpdate()
+{
+    static QString gitUserRepo;
+    if(gitUserRepo.isEmpty()) {
+        //https://github.com/user/repo.git
+        const static QRegularExpression regex("github.com/(.*).git");
```

Comment:
```suggestion
        const static QRegularExpression regex("github.com/(.*)");
```
I don't put `.git`, it's not needed here or there.

Reply:
the regex will capture both cases with my suggested edit.

Reply:
The problem is that "https://api.github.com/repos/bluerobotics/ping-viewer.git/releases" does not exist.